Added HB+NS notification UDA functions

gvarao opened the pull request.

sriram-rsankar commented on juniper_official/Paragon/Northstar/notify_northstar.py:
```diff
@@ -0,0 +1,188 @@
+
+#Queries the NorthStar Node ID using device hostname.     
+def get_node_id(hostname, **kwargs):
+    r = requests.get(node_get_url.format(hostname=hostname), headers=headers, verify=False)
```
Should we call requests.get inside try except block ?

gvarao replied:
Added try except block

sriram-rsankar commented on juniper_official/Paragon/Northstar/notify_northstar.py:
```diff
@@ -0,0 +1,188 @@
+    node_params = get_node_id(hostname)
+    if node_params:
+        node_id = node_params[0]
+        r = requests.get(link_get_url.format(node_id=node_id, if_name=ifd_name +"."+ ifl_number), headers=headers, verify=False)
```
Please catch the exception raised by request.get.

gvarao replied:
Added try except block

sriram-rsankar commented on juniper_official/Paragon/Northstar/notify_northstar.py:
```diff
@@ -0,0 +1,188 @@
+            print ('PUT  response='+str(c))
+        
+            d = requests.delete(mnt_url+'/'+str(mnt_index), headers=headers, verify=False)
+            print ('DELETE  response='+str(d))
```
Is this print needed

gvarao replied:
Removed print statements

gvarao commented:
Hi Sri Ram,

I have addressed all comments.

Regards,
Venkat(GV)

From: sriram-rsankar <notifications@github.com>
Reply to: Juniper/healthbot-rules <reply@reply.github.com>
Date: Thursday, 11 February 2021 at 11:30 AM
To: Juniper/healthbot-rules <healthbot-rules@noreply.github.com>
Cc: GV <gvenkata@juniper.net>, Author <author@noreply.github.com>
Subject: Re: [Juniper/healthbot-rules] Added HB+NS notification UDA functions (#492)

[External Email. Be cautious of content]


@sriram-rsankar commented on this pull request.

________________________________

In juniper_official/Paragon/Northstar/notify_northstar.py<https://github.com/Juniper/healthbot-rules/pull/492#discussion_r574265023>:

> +    r = requests.get(node_get_url.format(hostname=hostname), headers=headers, verify=False)

+    res = r.json()

+    if (r.status_code == 200) and (len(res)>0):

+        (node_id) = res[0]['id']

+        (node_index) = res[0]['nodeIndex']

+        return node_id, node_index

+    else:

+        raise Exception("Node doesn't exist")

+

+

+#Queries the NorthStar link ID using device hostname and IFD name and IFL number.

+def get_link_index(hostname, ifd_name, ifl_number, **kwargs):

+    node_params = get_node_id(hostname)

+    if node_params:

+        node_id = node_params[0]

+        r = requests.get(link_get_url.format(node_id=node_id, if_name=ifd_name +"."+ ifl_number), headers=headers, verify=False)

Please catch the exception raised by request.get.

________________________________

In juniper_official/Paragon/Northstar/notify_northstar.py<https://github.com/Juniper/healthbot-rules/pull/492#discussion_r574265767>:

> +    hostname = kwargs['device_id']

+    node_params = get_node_id(hostname)

+    if node_params:

+        node_id = node_params[0]

+        node_index = node_params[1]

+        r = requests.get(node_mnt_url.format(node_index=node_index), headers=headers, verify=False)

+        res = r.json()

+        if (r.status_code == 200) and (len(res)>0):

+            mnt_index = res[0]['maintenanceIndex']

+            res[0]['status']='completed'

+            res = json.dumps(res[0])

+            c = requests.put(mnt_url+'/'+str(mnt_index), data=res, headers=headers, verify=False)

+            print ('PUT  response='+str(c))

+

+            d = requests.delete(mnt_url+'/'+str(mnt_index), headers=headers, verify=False)

+            print ('DELETE  response='+str(d))

Is this print needed

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/Juniper/healthbot-rules/pull/492#pullrequestreview-588262190>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AD4JHVCPQ4OCIVLLJC5NSLDS6NW57ANCNFSM4XC7XBIQ>.


Juniper Business Use Only
